log4j-core/src/main/java/org/apache/logging/log4j/core/LoggerContext.java

```diff
@@ -812,6 +819,22 @@ private void initApiModule() {
                 .init(); // Or make public and call ThreadContext.init() which calls ThreadContextMapFactory.init().
     }
 
+    private Logger newInstance(final String name, final MessageFactory messageFactory) {
+        final Logger logger = newInstance(this, name, messageFactory);
+        final MessageFactory loggerMessageFactory = logger.getMessageFactory();
+        if (!loggerMessageFactory.equals(messageFactory)) {
+            StatusLogger.getLogger()
+                    .error(
+                            "Newly created logger with name `{}` and message factory `{}` was actually requested to be created with a different message factory: `{}`.\n"
+                                    + "This generally hints a problem.\n"
+                                    + "Please report this using the Log4j project issue tracker.",
+                            name,
+                            loggerMessageFactory,
+                            messageFactory);
```

Comment:
... instead of here, where it is basically impossible (unless the `newInstance` method is broken).

Reply:
Can we put this test in a test suite for `LoggerContext` implementations?

If `newInstance` is called with a given `name` and `messageFactory`, we need to assert that the returned logger has that name and message factory.

Reply:
Fixed in 8a89e8c3903d4a5836459f19c6500938fd89b219. Mind checking it again, please?